starlark/unpack.go

```diff
@@ -113,12 +128,20 @@ kwloop:
 	for _, item := range kwargs {
 		name, arg := item[0].(String), item[1]
 		for i := 0; i < nparams; i++ {
-			if paramName(pairs[2*i]) == string(name) {
+			pName, skipNone := paramName(pairs[2*i])
+			if pName == string(name) {
 				// found it
 				if defined.set(i) {
 					return fmt.Errorf("%s: got multiple values for keyword argument %s",
 						fnname, name)
 				}
+
+				if skipNone {
```

Comment:
For the record: setting `defined[i]` before the skipNone check appears to be consistent with Python3. In other words, `f(x=None, **dict(x=1))` gives a "duplicate named parameter" error. The x=None parameter is not completely ignored.